### PR TITLE
fix: infer output stream directories from namelist

### PR DESCRIPTION
**Pull request by leclairm**

### CODE REVIEW

**Comment by leclairm** [comment]
The way ICON builds the output path still needs to be tested before merging this PR. If the logic is correct, it was easy and we can still allow any specification in the ICON namelist with the plugin.

**Comment by DropD** [comment]
Thanks! I will merge a few other things, which are already tested first and then come back and test this after merging. If this is successful, maybe the solution in #4 will change to actually just handling the place holders for input file names in analogous ways.

File: src/aiida_icon/calculations.py [hunk 1]
```diff
@@ -60,8 +60,19 @@ def define(cls, spec: calcjob.CalcJobProcessSpec) -> None:  # type: ignore[overr
...
-            folder.get_subfolder(pathlib.Path(output_spec["output_filename"]).name, create=True)
+        def get_output_dir(out_stream_spec: f90nml.namelist.Namelist) -> pathlib.Path:
+            """Replicate ICON logic in forming the filenames to get the output dir"""
+
+            filename_format = out_stream_spec.get('filename_format', '<output_filename>_XXX_YYY')
+            output_filename = out_stream_spec.get('output_filename', '')
+            return pathlib.Path(filename_format.replace('<output_filename>', output_filename)).parent
+
+        # if only 1 output stream, make it a list
+        if type(specs := model_namelist_data["output_nml"]) is f90nml.namelist.Namelist:
```
Comment (DropD) [inline src/aiida_icon/calculations.py:71]:
```suggestion
        if isinstance(specs := model_namelist_data["output_nml"], f90nml.namelist.Namelist):
```

**Comment by leclairm** [comment]
I checked the mechanism and it works as expected in ICON so we can use that small PR to create necessary output directories.

**Review by DropD — DISMISSED** [review]

**Review by DropD — APPROVED** [review]
lgtm

**Pull request merged**